### ISSUE TITLE
Failed to inject docker in PREROUTING chain: …: No chain/target/match by that name.

### ISSUE THREAD

**Issue by devurandom** [docker/docker#13931]
On my system the docker daemon (v1.6.1) does not start. The message in the log is:
```
Failed to inject docker in PREROUTING chain: iptables failed: iptables --wait -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER: iptables: No chain/target/match by that name.
```

**Comment by devurandom** [comment]
```
$ docker version
Client version: 1.6.1
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 97cd073
OS/Arch (client): linux/amd64
FATA[0000] Get http:///var/run/docker.sock/v1.18/version: read unix /var/run/docker.sock: connection reset by peer. Are you trying to connect to a TLS-enabled daemon without TLS? 
```

```
$ docker info
FATA[0000] Cannot connect to the Docker daemon. Is 'docker -d' running on this host? 
```

```
$ uname -a
Linux ernie 4.0.5-gentoo #1 SMP Sun Jun 7 16:09:08 CEST 2015 x86_64 AMD Athlon(tm) 64 X2 Dual Core Processor 5000+ AuthenticAMD GNU/Linux
```

Provide additional environment details: I am running this on my desktop computer.

List the steps to reproduce the issue:
1. `systemctl start docker`

Describe the results you received:
```
$ sudo systemctl status docker
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib64/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sun 2015-06-14 19:10:14 CEST; 7s ago
     Docs: http://docs.docker.com
  Process: 24301 ExecStart=/usr/bin/docker -d -H fd:// (code=exited, status=1/FAILURE)
 Main PID: 24301 (code=exited, status=1/FAILURE)
```

Describe the results you expected:
As above, except:
```
   Active: active (running) since Sun 2015-06-14 19:10:14 CEST; 7s ago
```

Provide additional info you think is important: docker logged the message I posted in my original report.

**Comment by coolljt0725** [comment]
I think your kernel config may missing some config, such as `CONFIG_NF_NAT_IPV4 or  CONFIG_NF_NAT or CONFIG_NF_NAT_NEEDED`, use the `contrib/check-config.sh` in docker source code to check if the necessary config of your kernel are enable.

**Comment by devurandom** [comment]
Guess that's it, `XT_MATCH_ADDRTYPE` is not available in the non-advanced netfilter config:
```
info: reading kernel config from /boot/config-4.0.5-gentoo ...

Generally Necessary:
- cgroup hierarchy: properly mounted [/sys/fs/cgroup]
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_DEVPTS_MULTIPLE_INSTANCES: enabled
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MACVLAN: enabled
- CONFIG_VETH: enabled
- CONFIG_BRIDGE: enabled
- CONFIG_BRIDGE_NETFILTER: missing
- CONFIG_NF_NAT_IPV4: enabled (as module)
- CONFIG_IP_NF_FILTER: enabled (as module)
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: missing
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled (as module)
- CONFIG_NF_NAT: enabled (as module)
- CONFIG_NF_NAT_NEEDED: enabled
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_MEMCG_SWAP: enabled
- CONFIG_MEMCG_SWAP_ENABLED: enabled
- CONFIG_BLK_CGROUP: enabled
- CONFIG_IOSCHED_CFQ: enabled
- CONFIG_CGROUP_PERF: enabled
- CONFIG_CFS_BANDWIDTH: missing
- Storage Drivers:
  - "aufs":
    - CONFIG_AUFS_FS: missing
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "btrfs":
    - CONFIG_BTRFS_FS: enabled
  - "devicemapper":
    - CONFIG_BLK_DEV_DM: enabled
    - CONFIG_DM_THIN_PROVISIONING: enabled
    - CONFIG_EXT4_FS: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "overlay":
    - CONFIG_OVERLAY_FS: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
  - "zfs":
    - /dev/zfs: missing
    - zfs command: missing
    - zpool command: missing
```

**Comment by crosbymichael** [comment]
This has been fixed by #16038

thanks!
